wgpu-native/src/instance.rs

```diff
@@ -88,15 +88,53 @@ pub fn wgpu_create_gl_instance(windowed_context: back::glutin::RawContext<back::
     HUB.surfaces.register_local(surface, &mut Token::root())
 }
 
-#[cfg(all(feature = "local", feature = "winit", not(feature = "glutin")))]
-#[no_mangle]
-pub extern "C" fn wgpu_instance_create_surface_from_winit(
+#[cfg(all(feature = "local", not(feature = "glutin")))]
+pub fn wgpu_instance_create_surface(
     instance_id: InstanceId,
-    window: &winit::Window,
+    raw_handle: raw_window_handle::RawWindowHandle,
 ) -> SurfaceId {
     let mut token = Token::root();
     let (instance_guard, mut token) = HUB.instances.read(&mut token);
-    let raw = instance_guard[instance_id].create_surface(window);
+    let instance = &instance_guard[instance_id];
+    let raw = match raw_handle {
+        #[cfg(all(target_os = "ios", feature = "gfx-backend-metal"))]
+        raw_window_handle::RawWindowHandle::IOS(h) =>
+            instance.create_surface_from_uiview(raw.ui_view, cfg!(debug_assertions)),
+        #[cfg(all(target_os = "ios", feature = "gfx-backend-metal"))]
+        raw_window_handle::RawWindowHandle::MacOS(h) =>
```

Comment:
Did we also want to support gfx-backend-vulkan on macOS? (I don't think we need to block the PR on this, but probably worth mentioning since we added it to gfx-hal)